src/include/network/routines.rb

```diff
@@ -513,138 +379,81 @@ def ReadHardware(hwtype)
-          drivers = Builtins.filter(Ops.get_list(card, "drivers", [])) do |d|
-            # ignoring more modules per driver...
-            module0 = Ops.get_list(d, ["modules", 0], []) # [module, options]
-            brk = broken_modules.include?(module0[0])
-
-            if brk
-              Builtins.y2milestone("In BrokenModules, skipping: %1", module0)
-            end
+          one["drivers"] = drivers
+
+          driver = drivers[0] || {}
+          one["active"] = driver["active"] || false
+          module0 = driver.fetch("modules", []).fetch(0, [])
+          one["module"] = module0[0] || ""
+          one["options"] = module0[1] || ""
+        end
 
-            !brk
-          end
-
-          if drivers == []
-            Builtins.y2milestone("No good drivers found")
-          else
-            one["drivers"] = drivers
-
-            driver = drivers[0] || {}
-            one["active"] = driver["active"] || false
-            module0 = Ops.get_list(driver, ["modules", 0], [])
-            one["module"] = module0[0] || ""
-            one["options"] = module0[1] || ""
-          end
-
-          # FIXME: this should be also done for modems and others
-          # FIXME: #13571
-          hp = card["hotplug"] || ""
-          if hp == "pcmcia" || hp == "cardbus"
-            one["hotplug"] = "pcmcia"
-          elsif hp == "usb"
-            one["hotplug"] = "usb"
-          end
-
-          # store the BUS type
-          bus = card["bus_hwcfg"] || card["bus"] || ""
-
-          if bus == "PCI"
-            bus = "pci"
-          elsif bus == "USB"
-            bus = "usb"
-          elsif bus == "Virtual IO"
-            bus = "vio"
-          end
-
-          one["bus"] = bus
-          one["busid"] = card["sysfs_bus_id"] || ""
-          one["mac"] = Ops.get_string(resource, ["hwaddr", 0, "addr"], "")
-          # is the cable plugged in? nil = don't know
-          one["link"] = Ops.get(resource, ["link", 0, "state"])
-
-          # Wireless Card Features
-          one["wl_channels"] = Ops.get(resource, ["wlan", 0, "channels"])
-          one["wl_bitrates"] = Ops.get(resource, ["wlan", 0, "bitrates"])
-          one["wl_auth_modes"] = Ops.get(resource, ["wlan", 0, "auth_modes"])
-          one["wl_enc_modes"] = Ops.get(resource, ["wlan", 0, "enc_modes"])
+        # FIXME: this should be also done for modems and others
+        # FIXME: #13571
+        hp = card["hotplug"] || ""
+        if hp == "pcmcia" || hp == "cardbus"
+          one["hotplug"] = "pcmcia"
+        elsif hp == "usb"
+          one["hotplug"] = "usb"
         end
 
+        # store the BUS type
+        bus = card["bus_hwcfg"] || card["bus"] || ""
+
+        if bus == "PCI"
+          bus = "pci"
+        elsif bus == "USB"
+          bus = "usb"
+        elsif bus == "Virtual IO"
+          bus = "vio"
+        end
+
+        one["bus"] = bus
+        one["busid"] = card["sysfs_bus_id"] || ""
+        one["mac"] = resource["hwaddr"][0]["addr"] || ""
```

Comment:
Yeah, maybe i was too optimistic. I tried to replace the Ops dependency

Ops.get_string(resource, ["hwaddr", 0, "addr"], "")

And it take cares of nil values, so i not realized this. 

Ruby 2.3 #dig wille be nice here ;)